simpeg/electromagnetics/time_domain/sources.py

```diff
@@ -1908,37 +1914,70 @@ def jInitialDeriv(self, simulation, v=None, adjoint=False, f=None):
         )
 
     def _getAmmr(self, simulation):
-        if simulation._formulation != "HJ":
+        if simulation._formulation == "EB":
             raise NotImplementedError
-
-        vol = simulation.mesh.cell_volumes
-        Div = sdiag(vol) * simulation.mesh.face_divergence
-        return (
-            simulation.mesh.edge_curl
-            * simulation.MeMuI
-            * simulation.mesh.edge_curl.T.tocsr()
-            - Div.T.tocsr()
-            * sdiag(1.0 / vol * simulation.mui)
-            * Div  # stabalizing term. See (Chen, Haber & Oldenburg 2002)
-        )
+        elif simulation._formulation == "HJ":
+            vol = simulation.mesh.cell_volumes
+            Div = sdiag(vol) * simulation.mesh.face_divergence
+            return (
+                simulation.mesh.edge_curl
+                * simulation.MeMuI
+                * simulation.mesh.edge_curl.T.tocsr()
+                - Div.T.tocsr()
+                * sdiag(1.0 / vol * simulation.mui)
+                * Div  # stabalizing term. See (Chen, Haber & Oldenburg 2002)
+            )
 
     def _aInitial(self, simulation):
-        A = self._getAmmr(simulation)
-        Ainv = simulation.solver(A)  # todo: store this
-        s_e = self.s_e(simulation, 0)
-        rhs = s_e + self.jInitial(simulation)
-        return Ainv * rhs
+        if self.srcType == "inductive":
+            # for an inductive source, use the Biot Savart law (from geoana) to compute the vector potential
+            if simulation._formulation == "EB":
+                vector_potential = np.zeros(simulation.mesh.n_edges)
+
+                for i in range(self.location.shape[0] - 1):
+                    line_current = LineCurrentWholeSpace(self.location[i : i + 2, :])
```

Comment:
Another note, I believe you should be able to pass the full path of the wire at once, instead of adding the segments individually.